xilem_core/src/views/memoize.rs

```diff
@@ -1,65 +1,68 @@
 // Copyright 2024 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
+use core::marker::PhantomData;
+
 use crate::{MessageResult, Mut, View, ViewId, ViewPathTracker};
 
 /// A view which supports Memoization.
 ///
 /// The story of Memoization in Xilem is still being worked out,
 /// so the details of this view might change.
-pub struct Memoize<D, F> {
-    data: D,
-    child_cb: F,
+pub struct Memoize<Data, InitView, State, Action> {
+    data: Data,
+    init_view: InitView,
+    phantom: PhantomData<fn() -> (State, Action)>,
 }
 
-pub struct MemoizeState<State, Action, Context, Message, V>
-where
-    Context: ViewPathTracker,
-    V: View<State, Action, Context, Message>,
-{
+pub struct MemoizeState<V, VState> {
     view: V,
-    view_state: V::ViewState,
+    view_state: VState,
     dirty: bool,
 }
 
-impl<D, V, F> Memoize<D, F>
+impl<Data, V, InitView, State, Action> Memoize<Data, InitView, State, Action>
 where
-    F: Fn(&D) -> V,
+    InitView: Fn(&Data) -> V,
 {
-    const ASSERT_CONTEXTLESS_FN: () = {
-        assert!(
-            core::mem::size_of::<F>() == 0,
-            "
+    /// Create a new `Memoize` view.
+    pub fn new(data: Data, init_view: InitView) -> Self {
```

Comment:
I don't think this should be `pub`?

Reply:
I guess we could also remove this entirely und just use a struct initializer in `memoize` as I think we settled for the `function()` syntax for views.